src/NetscriptFunctions/CodingContract.ts

```diff
@@ -34,20 +33,11 @@ export function NetscriptCodingContract(player: IPlayer, workerScript: WorkerScr
         const hostname = ctx.helper.string("hostname", _hostname);
         const contract = getCodingContract(ctx, "attempt", hostname, filename);
 
-        // Convert answer to string. If the answer is a 2D array, then we have to
-        // manually add brackets for the inner arrays
-        let answerStr = "";
-        if (is2DArray(answer)) {
-          const answerComponents = [];
-          for (let i = 0; i < answer.length; ++i) {
-            answerComponents.push(["[", String(answer[i]), "]"].join(""));
-          }
-
-          answerStr = answerComponents.join(",");
-        } else {
-          answerStr = String(answer);
-        }
+        if (typeof answer !== "number" && typeof answer !== "string" && !Array.isArray(answer))
```

Comment:
typechecking to actually enforce the input types allowed in documentation, and provide a helpful error message.